src/analyzer/analyzerwaveform.cpp

```diff
@@ -23,7 +25,9 @@ AnalyzerWaveform::AnalyzerWaveform(UserSettingsPointer pConfig) :
     m_filter[1] = 0;
     m_filter[2] = 0;
 
+    QMutexLocker lock(&s_mutex);
```

Comment:
Mostly ignorance about the correct ways to use QT. I am changing it right now and doing a quick test.